host_modules/systemd_service.py

```diff
@@ -0,0 +1,45 @@
+"""Systemd service handler"""
+
+from host_modules import host_service
+import subprocess
+
+MOD_NAME = 'systemd'
+ALLOWED_SERVICES = ['snmp', 'swss', 'dhcp_relay', 'radv', 'restapi', 'lldp', 'sshd', 'pmon', 'rsyslog']
+EXIT_FAILURE = 1
+
+
+class SystemdService(host_service.HostModule):
+    """
+    DBus endpoint that executes the service command
+    """
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def restart_service(self, service):
+        if not service:
+            return EXIT_FAILURE, "Dbus restart_service called with no service specified"
+        if service not in ALLOWED_SERVICES:
+            return EXIT_FAILURE, "Dbus does not support {} service restart".format(service)
+
+        cmd = ['/usr/bin/systemctl', 'reset-failed', service]
+        result = subprocess.run(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
```

Comment:
>result [](http://example.com/codeflow?start=8&length=6)

Why ignore checking `result`?

Reply:
If `snmp` was intentionally stopped, `reset-failed` throws error irrelevant to us: `Failed to reset failed state of unit snmp.service: Unit snmp.service not loaded.`

For example: 
```
admin@str-s6100-acs-1:~$ systemctl status snmp
â— snmp.service - SNMP container
     Loaded: loaded (/lib/systemd/system/snmp.service; static)
     Active: active (running) since Wed 2024-05-22 01:24:48 UTC; 4s ago
   Main PID: 5143 (snmp.sh)
      Tasks: 3 (limit: 9456)
     Memory: 30.1M
     CGroup: /system.slice/snmp.service
             â”œâ”€5143 /bin/bash /usr/local/bin/snmp.sh wait
             â”œâ”€5147 /bin/bash /usr/bin/snmp.sh wait
             â””â”€5148 python3 /usr/local/bin/container wait snmp
admin@str-s6100-acs-1:~$ sudo systemctl reset-failed snmp
admin@str-s6100-acs-1:~$ sudo systemctl stop snmp
admin@str-s6100-acs-1:~$ systemctl status snmp
â—‹ snmp.service - SNMP container
     Loaded: loaded (/lib/systemd/system/snmp.service; static)
    Drop-In: /etc/systemd/system/snmp.service.d
             â””â”€auto_restart.conf
     Active: inactive (dead)
admin@str-s6100-acs-1:~$ sudo systemctl reset-failed snmp
Failed to reset failed state of unit snmp.service: Unit snmp.service not loaded.
```

Reply:
Thanks, that makes sense. I added a check to ignore the known expected error